rq/worker.py

```diff
@@ -883,14 +893,19 @@ def handle_job_failure(self, job, queue, started_job_registry=None,
                     job_class=self.job_class
                 )
             job.worker_name = None
-            # Requeue/reschedule if retry is configured
-            if job.retries_left and job.retries_left > 0:
-                retry = True
-                retry_interval = job.get_retry_interval()
-                job.retries_left = job.retries_left - 1
+
+            # check whether a job was stopped intentionally and set the job
+            # status appropriately if it was this job.
+            job_is_stopped = self._stopped_job_id and self._stopped_job_id == job.id
```

Comment:
I think we can do `job_is_stopped = self._stopped_job_id == job.id` here, since if `_stopped_job_id` is `None`, it would evaluate to `False` anyway.